sdks/python/apache_beam/runners/interactive/caching/streaming_cache_test.py

```diff
@@ -0,0 +1,167 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import absolute_import
+
+import unittest
+
+from google.protobuf import timestamp_pb2
+
+from apache_beam import coders
+from apache_beam.portability.api.beam_interactive_api_pb2 import InteractiveStreamHeader
+from apache_beam.portability.api.beam_interactive_api_pb2 import InteractiveStreamRecord
+from apache_beam.portability.api.beam_runner_api_pb2 import TestStreamPayload
+from apache_beam.runners.interactive.caching.streaming_cache import StreamingCache
+from apache_beam.utils import timestamp
+
+
+def to_timestamp_proto(timestamp_secs):
+  """Converts seconds since epoch to a google.protobuf.Timestamp.
+  """
+  seconds = int(timestamp_secs)
+  nanos = int((timestamp_secs - seconds) * 10**9)
+  return timestamp_pb2.Timestamp(seconds=seconds, nanos=nanos)
+
+
+class InMemoryReader(object):
+  def __init__(self, tag=None):
+    self._records = [InteractiveStreamHeader(tag=tag).SerializeToString()]
```

Comment:
It'd be cleaner if the stream returned proto objects, rather than the stream returning strings which need to be parsed in a certain way.

Reply:
The code for the underlying PCollection Cache isn't merged yet, and I haven't decided if I want to create a new subclass or use it directly. This code assumes that the PCollection cache will just writes strings to a file. I can change it later once the PCollection Cache code is merged.

Reply:
The point stands whether or not we'll merge with PCollectionCache later.

Reply:
I'm hoping that we will have the PCollectionCache! Regardless, In the meantime I changed it to return proto objects instead of the string.